src/app/main.rs

```diff
@@ -252,31 +254,25 @@ where
             // settings. Stack layers above this operate on an `Endpoint` with
             // the TLS client config is marked as `NoConfig` when the endpoint
             // has a TLS identity.
-            let router_layer = router::Layer::new(outbound::Recognize::new())
+            let router_stack = router::Layer::new(Recognize::new())
                 .and_then(limit::Layer::new(MAX_IN_FLIGHT))
                 .and_then(timeout::Layer::new(config.bind_timeout))
                 .and_then(buffer::Layer::new())
-                .and_then(balance::Layer::new(outbound::discovery::Resolve::new(
-                    resolver,
-                )))
-                .and_then(outbound::orig_proto_upgrade::Layer::new())
-                .and_then(outbound::tls_config::Layer::new(tls_client_config))
-                .and_then(proxy::http::metrics::Layer::new(
-                    http_metrics,
-                    classify::Classify,
-                ))
+                .and_then(balance::layer())
+                .and_then(resolve::layer(Resolve::new(resolver)))
```

Comment:
Very minor, but: 👍 on changing these to free functions! definitely more readable!